department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-15239

Resolves [APPEALS-15239](https://vajira.max.gov/browse/APPEALS-15239)

### Description
Created a rails migration to add the **completed_by** relationship to the task model. Also, displayed this value on the Case Timeline when a Letter Task is completed. 

### Acceptance Criteria
- [ ] The task model has the **completed_by** relation to the **users** model
- [ ] The **tasks** table has a **completed_by_id**
- [ ] When completing an Initial Letter Task, the **COMPLETED BY** user's CSS ID shows in the Case Timeline 
- [ ] When completing a Post Initial Letter Task, the **COMPLETED BY** user's CSS ID shows in the Case Timeline 
- [ ] When completing a Final Letter Task, the **COMPLETED BY** user's CSS ID shows in the Case Timeline 

### Testing Plan
1. To create a contested claim with an initial letter task, go to Caseflow and sign in as a BVA Intake user like **BVADWISE**. If you already have test data, skip to step **7.**
2. Navigate to the **Intake** app.
3. Select the first option from the dropdown and click continue
![image](https://user-images.githubusercontent.com/99915461/220765144-2790b46d-7535-4768-b193-baaf1a5083fa.png)

4. Enter a veteran's file number from the database.
![image](https://user-images.githubusercontent.com/99915461/220765206-0b24c1b5-05f8-4731-a389-a9a3fb01caf8.png)

5. Fill out the form and click continue.
![image](https://user-images.githubusercontent.com/99915461/220765294-415c5482-bcef-4861-8a7f-fa37543c3860.png)

6. Click **add issue** and select a contested claim issue. Click **add this issue** followed by **establish appeal** to complete the intake. You should be greeted with a green success banner.
![image](https://user-images.githubusercontent.com/99915461/220765454-8213d0e1-b72b-4e23-977a-564db428c55e.png)
![image](https://user-images.githubusercontent.com/99915461/220765492-65fdc01b-2ce9-4e62-a18e-e39d98404440.png)
![image](https://user-images.githubusercontent.com/99915461/220765517-aa88c128-0473-48c9-8fa9-cd64dd2a1363.png)

7. Sign in as a **COB** user, like **COB_USER**. Navigate to the **Queue** app.
8. Click the **switch views** dropdown followed by the **Clerk of the Board** option.
![image](https://user-images.githubusercontent.com/99915461/220766010-9a2d9dd7-c77f-4113-8362-4d98a8209dc3.png)

9. Select the **Send Initial Task** from the table.
![image](https://user-images.githubusercontent.com/99915461/220766140-d0b0b3c8-df10-4a5d-bf18-4f9fe43edd43.png)

10. On the case details page, click the **actions** dropdown and **mark task as complete**.
![image](https://user-images.githubusercontent.com/99915461/220766235-b0ba3f2b-fda5-4f5c-ba55-ae087cd27950.png)

11. Select the option for **45 Days** and **mark as complete**. 
![image](https://user-images.githubusercontent.com/99915461/220766320-3eb6dc38-60c7-46d0-9543-8b63d672f7c3.png)

12. You should be greeted by a green success banner. 
![image](https://user-images.githubusercontent.com/99915461/220766390-27e99ac1-53a8-4c33-bb9f-6e5bdcde8845.png)

13. Click the back arrow to return to the Case Detail page. Navigate to the bottom of the page to look at the Case Timeline. You should see the **Initial Letter Task** has been successfully completed, and the **COMPLETED BY** should be the **COB_USER** you were signed in as.
![image](https://user-images.githubusercontent.com/99915461/220766680-3657ee41-7452-45e5-8953-9edcd4a432e7.png)

14. Scroll back up to the top of the page and click on the **actions** dropdown. Click the option **proceed to final notification letter**. Fill out the instructions on the next form and click **Proceed to final letter**.
![image](https://user-images.githubusercontent.com/99915461/220766972-3cbf3ef9-4eb4-490e-afea-df3a527a9db0.png)
![image](https://user-images.githubusercontent.com/99915461/220767105-d2ba6652-6f59-43d7-8d7c-113272b350a8.png)

15. You should be greeted with a green success banner.
![image](https://user-images.githubusercontent.com/99915461/220767197-278be1d1-4711-4b47-9308-95befc356e3b.png)

16. Click the back arrow to navigate back to the Case Details page. Scroll down to the bottom of the page. The **Post Initial Task** should be successfully completed and show the same **COB_USER** for **COMPLETED BY**.
![image](https://user-images.githubusercontent.com/99915461/220767576-79d13f75-30a8-48a0-8cba-093f4134d219.png)
